jwst/resample/resample.py

```diff
@@ -54,10 +54,11 @@ def __init__(self, input_models, output=None, **pars):
         # Define output WCS based on all inputs, including a reference WCS
         self.output_wcs = resample_utils.make_output_wcs(self.input_models)
         log.debug('Output mosaic size: {}'.format(self.output_wcs.data_size))
-        self.blank_output = datamodels.DrizProductModel(self.output_wcs.data_size)
+        self.blank_output = datamodels.ImageModel(self.output_wcs.data_size)
 
         # update meta data and wcs
-        self.blank_output.update(input_models[0])
+        self.blank_output.update(input_models[0], only='PRIMARY')
+        self.blank_output.update(input_models[0], only='SCI')
```

Comment:
So the case of regular imaging data seems to not respond properly to this change. I'd go ahead and keep these new lines in there, but in addition we need to make a change in `resample_step.py` to add the following lines right after line 77:
```
  if hasattr(model.meta, "bunit_err") and model.meta.bunit_err is not None:
      del model.meta.bunit_err
```
Not elegant, but it finally gets rid of the empty ERR extensions in i2d products and hence fixes the current problems with several of the imaging-mode reg tests.